design/resource-constraints.md

```diff
@@ -0,0 +1,52 @@
+# Resource Constraints
+
+## Overview
+Resource Constraints allow the rook components to be in specific Kubernetes Quality of Service (QoS) classes. For this the components started by rook need to have resource requests and/or limits set depending on which class the component(s) should be in.
+
+Ceph has recommendations for CPU and memory for each component. The recommendations can be found here: http://docs.ceph.com/docs/master/start/hardware-recommendations/
+
+## Application of Resource Constraints
+The resource constraints are defined in the rook Cluster, Filesystem and RGW CRDs.
+
+### User defined
+User defined values, when given, are above the calculated values.
+
+#### Global User Resource Specification
+A Kubernetes resource requirement object looks like this:
+```yaml
+requests:
+  cpu: “2”
+  memory: “1Gi”
+limits:
+  cpu: “3”
+  memory: “2Gi”
+```
+
+The key in the CRDs to set resource requirements is named `resources`.
+The following components will allow a resource requirement to be set for:
+* `api`
+* `mds`
+* `mgr`
+* `mon`
+* `osd`
```

Comment:
is there an repercussion if the osd is trying to consume more than the limit? If that happens, kubernetes will terminate it. This could result in a degraded cluster.

Reply:
For memory, which is not elastic, a pod can be killed if it goes over the limit.  CPU will just be throttled.